clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,458 @@
+    };
+    let updated_config = match serde_json::from_str::<TeliodDaemonConfigPartial>(body) {
+        Ok(updates) => updates,
+        Err(e) => {
+            eprintln!("Error parsing config json: {}", e);
+            return text_response(StatusCode::BAD_REQUEST, "Invalid JSON payload");
+        }
+    };
+
+    config.update(updated_config);
+
+    match fs::write(TELIOD_CFG, serde_json::to_string_pretty(&config).unwrap()) {
+        Ok(_) => text_response(StatusCode::OK, "Configuration updated successfully"),
+        Err(_) => text_response(
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Failed to write updated config",
+        ),
+    }
+}
+
+fn get_status() -> Response {
+    if !is_teliod_running() {
+        return text_response(StatusCode::GONE, "Application is not running.");
+    }
+
+    match teliod_blocking_query!(ClientCmd::GetStatus) {
+        Ok(Ok(daemon_reply)) => match CommandResponse::deserialize(&daemon_reply) {
+            Ok(CommandResponse::StatusReport(status)) => text_response(
+                StatusCode::OK,
+                serde_json::to_string_pretty(&status).unwrap_or_default(),
+            ),
+            Ok(cmd_response) => text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Unexpected Teliod response: {}", cmd_response.serialize()),
+            ),
+            Err(error) => text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to retrieve status: {}", error),
+            ),
+        },
+        Ok(Err(error)) => text_response(
+            StatusCode::GONE,
+            format!("Failed to communicate with the daemon: {}", error),
+        ),
+        Err(error) => text_response(
+            StatusCode::GATEWAY_TIMEOUT,
+            format!("Failed to communicate with the daemon: {}", error),
+        ),
+    }
+}
+
+fn get_teliod_logs() -> Response {
+    match fs::read_to_string(TELIOD_LOG) {
+        Ok(logs) => text_response(StatusCode::OK, logs),
+        Err(error) => text_response(
+            StatusCode::BAD_GATEWAY,
+            format!("Error reading teliod log file: {}", error),
+        ),
+    }
+}
+
+fn get_meshnet_logs() -> Response {
+    match fs::read_to_string(MESHNET_LOG) {
+        Ok(logs) => text_response(StatusCode::OK, logs),
+        Err(error) => text_response(
+            StatusCode::BAD_GATEWAY,
+            format!("Error reading meshnet log file: {}", error),
+        ),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{fs, num::NonZeroU64, path::PathBuf, sync::Mutex};
+
+    use lazy_static::lazy_static;
+    use reqwest::StatusCode;
+    use tracing::level_filters::LevelFilter;
+    use uuid::Uuid;
+
+    use super::{update_config, TeliodDaemonConfig};
+    use crate::{
+        config::{InterfaceConfig, MqttConfig, Percentage},
+        configure_interface::InterfaceConfigurationProvider,
+    };
+
+    pub const TELIOD_CFG: &str = "/tmp/teliod_config.json";
```

Comment:
I think this will fail on windows - can you change it to use https://crates.io/crates/tempfile ?

Reply:
I think this is out of scope for now, we can make this as an issue, to make cgi standalone and cross platform

Reply:
This is not about making cgi cross platform - it's about making unit tests pass for developer(s?) on windows.

Reply:
How would windows devs run these unit tests?

Reply:
Hm, you right - there is no configurability in this module and all paths are completely hardcoded. It's probably relatively cheap to make paths be configurable with a prefix (which would be empty in production) and set it to temporary dir in tests. But fell free to treat this thread as nitpick and ignore it if you want.